pontoon/sync/core.py

```diff
@@ -69,15 +71,48 @@ def collect_entities(db_project, vcs_project):
     When a match isn't found, the missing entity will be None.
     """
     vcs_entities = get_vcs_entities(vcs_project)
+    db_entities = get_db_entities(db_project, vcs_project.changed_files)
+    entity_keys = set().union(db_entities.keys(), vcs_entities.keys())
+
+    for key in entity_keys:
+        yield key, db_entities.get(key, None), vcs_entities.get(key, None)
+
+
+def get_changed_db_resources(db_project):
+    """
+    Returns a map of resource paths and their locales that where changed from the last sync.
+    """
+    resources = defaultdict(set)
+    changes = (
+        ChangedEntityLocale.objects
+            .filter(entity__resource__project=db_project)
+            .prefetch_related('locale', 'entity__resource')
+    )
+
+    for change in changes:
+        resources[change.entity.resource.path].add(change.locale)
+
+    return resources
+
+
+def collect_translation_entities(db_project, vcs_project):
```

Comment:
This is almost the same as `collect_source_entities`, but with:
* call to update `vcs_project.changed_files`
* `get_db_entities` is called without the `vcs_project.changed_files` parameter (why?)

We could probably reuse some code here.